buildSrc/src/main/groovy/org/apache/beam/gradle/VendorJavaPlugin.groovy

```diff
@@ -131,15 +131,15 @@ artifactId=${project.name}
         inputs.files project.configurations.shadow.artifacts.files
         doLast {
           project.configurations.shadow.artifacts.files.each {
-            FileTree exposedClasses = project.zipTree(it).matching {
+            FileTree unexpectedlyExposedClasses = project.zipTree(it).matching {
               include "**/*.class"
               exclude "org/apache/beam/vendor/**"
               // BEAM-5919: Exclude paths for Java 9 multi-release jars.
-              exclude "**/module-info.class"
```

Comment:
The glob matching semantics [allegedly](https://docs.gradle.org/current/javadoc/org/gradle/api/tasks/AbstractCopyTask.html#include-java.lang.String...-) follow [ANT pattern matching](https://ant.apache.org/manual/dirtasks.html#patterns) so I'm lost as to why `**/module-info.class` didn't exclude the top level pattern.

Reply:
`**/module-info.class` is excluding the top level pattern. If I understand it correctly, the problem is we only want the top level one to be excluded. This was a regression introduced [here](https://github.com/apache/beam/pull/10463/files#diff-23833058cbf2c1172b90e7764032aa59R959) during the last grpc version change.

Reply:
Now I understand, misread what the exclusion was for. Thought it was for excluding the file and not for excluding files during validation.